lib/puppet/pops/serialization/to_data_converter.rb

```diff
@@ -92,7 +100,11 @@ def to_data(value)
         elsif @rich_data
           { PCORE_TYPE_KEY => PCORE_TYPE_SYMBOL, PCORE_VALUE_KEY => value.to_s }
         else
-          unknown_to_string_with_warning(value)
+          if @force_symbol
+            value
+          else
+            unknown_to_string_with_warning(value)
```

Comment:
It's probably outside the scope of this PR, but since we added the new silence warnings option, we can also use it here.